src/Network/GRPC/MQTT/Message/Packet.hs

```diff
@@ -371,4 +412,111 @@ isMissingPackets = do
       Nothing -> pure True
       Just ct -> do
         len <- lengthPacketSet pxs
-        pure (fromIntegral len - 1 < ct)
\ No newline at end of file
+        pure (fromIntegral len - 1 < ct)
+
+-- Unsafe ----------------------------------------------------------------------
+
+fromPacketBuildR ::
+  MonadIO m =>
+  Ptr Word8 ->
+  Int ->
+  Packet ByteString ->
+  m (Ptr Word8, Int)
+fromPacketBuildR buffer unused packet = do
+  let build = Encode.reverseMessageBuilder (wireBuildPacket packet)
+  liftIO (fromBuildR build buffer unused)
```

Comment:
Comment only: I'm not saying this is a better alternative, but given the simplicity of `Packet` serialization there is one other way to handle this situation: directly construct the encoding by using `ST` actions on a `MutableByteArray#` and a knowledge of proto3 format, instead of by using special access to reverse-builder internals.  That would be less fragile when the builder changes, but would require a careful manual application of the proto3 specification.  Execution speed would likely be comparable between the two approaches.

Reply:
I think this is a good idea, but outside the scope of these changes.

Reply:
Personally I'm not sure if it would be better or worse, overall; it's just an option I'm mentioning.